### ISSUE TITLE
Enhancement: Better argument type information

### ISSUE THREAD

**Issue by rtek** [rollbar/rollbar-php#385]
Traces that contain arguments of objects show up in Rollbar as the json_encoded version of the object which usually does not contain much information. Many objects do not implement \JsonSerializable nor have public properties so the argument is serialized as `{}`.

Can there be a mechanism to include type information such that the argument traces can be shown with more richness?

This is also true for associative arrays - there is no way to differentiate between an associative array argument or an object since they both encode to json objects.

A custom transformer goes a long way to address.

```php
class Transformer implements TransformerInterface
{
    public function transform(Payload $payload, $level, $toLog, $context)
    {
        $value = $payload->getData()->getBody()->getValue();
        if($value instanceof Trace) {
            /** @var Frame[] $frames */
            $frames = $value->getFrames();
            foreach($frames as $frame) {
                //make a copy of args since they may be by ref
                $args = [];
                foreach((array)$frame->getArgs() as $arg) {
                    if(is_object($arg)) {
                        $args[] = [
                            'class' => get_class($arg),
                            'value' => $arg,
                        ];
                    } else {
                        $args[] = $arg;
                    }
                }
                $frame->setArgs($args);
            }
        }

        return $payload;
    }
}
```
For example:

Before transformer
```
PHPUnit\Framework\TestSuite::run arguments:
--
0 | {}
```

After transformer
```
PHPUnit\Framework\TestSuite::run arguments:
0 | {
"class": "PHPUnit\\Framework\\TestResult",
"value": {}
}
```

**Comment by rivkahstandig3636** [comment]
Thoughts @ArturMoczulski?

**Comment by ArturMoczulski** [comment]
Thank you for bringing this up @rtek It's definitely a good idea for improvement. I'll add your suggestions to our workstack.

**Comment by ArturMoczulski** [comment]
Done. It will be included in `v1.6.0`
